lms/models/notes.py

```diff
@@ -0,0 +1,58 @@
+from typing import Optional
+
+from flask import jsonify, request
+from flask_login import current_user
+
+from lms.lmsdb.models import CommentText, Exercise, Note, NotePrivacy, User
+from lms.models.errors import fail
+
+
+def create_note_instance(
+    creator: User,
+    user: User,
+    note_text: CommentText,
+    privacy: NotePrivacy,
+    exercise: Optional[Exercise] = None,
+) -> Note:
```

Comment:
I'd be suspicious here and check if the user have the right permissions. (You did it on `delete_note`, so there is no reason not to do it here as well)

Reply:
I'm not really using it to create a Note.
Just for the conftest. Are you sure we need to add this check?

Reply:
Just deleted this method